app/src/main/java/com/eveningoutpost/dexdrip/insulin/LinearTrapezoidInsulin.java

```diff
@@ -0,0 +1,56 @@
+package com.eveningoutpost.dexdrip.insulin;
+
+import com.google.gson.JsonObject;
+
+import java.util.ArrayList;
+
+public class LinearTrapezoidInsulin extends Insulin {
+    /// curvedata and all timestamps are defined in minutes
+    private long onset;  // when does profile activity starts
+    private long t1;     // when does activity reaches max
+    private long t2;     // when does activity leaves max
+    private long t3;     // when does activity ends
+
+    private double max;
+
+    public LinearTrapezoidInsulin(String n, String dn, ArrayList<String> ppn, String c, JsonObject curveData) {
+        super(n, dn, ppn, c, curveData);
+
+        onset = curveData.get("onset").getAsLong();
+        if (curveData.get("peak").getAsString().contains("-")) {
```

Comment:
because there aren't multiple peaks - its just a linear trapezoid: an onset; a ramp up, a peak time and at the end a ramp down to 0. Sum under this curve is exactly the injected dose

Reply:
Semantics. There are multiple points in the trapezoid and you've chosen to put two of the points into one field in the datastructure and then extract them later rather than just store them separately.